### PR TITLE
Eli/swap permit

### LINKED ISSUE #200: Swap contract with permited Accept entrypoint

### PR DESCRIPTION
**Pull request by EGuenz**
Closes #200

### CODE REVIEW

**Review by michaeljklein — CHANGES_REQUESTED** [review]

File: packages/minter-contracts/ligo/src/swaps/fa2_allowlisted_swap_offchain_nocounter.mligo [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    let ops = accept_swap_update_ops_list(swap, swap_accepter, ops, storage.swap) in
+    let swap_storage = accept_swap_update_storage(p.swap_id, swap, swap_accepter, storage.swap) in
+    (ops, {storage with swap = swap_storage})
+  end
+
+let rec offchain_accept_batch_helper(ps, (ops, storage) : permit_accept_param list * (operation list * storage)) 
+  : allowlist_return = 
+    match ps with 
+      | [] -> (ops, storage)
+      | permit :: remaining_permits ->
```
Comment (michaeljklein) [inline packages/minter-contracts/ligo/src/swaps/fa2_allowlisted_swap_offchain_nocounter.mligo:31]: would it be straightforward to replace this with a fold over the list? it gets compiled to a `LOOP_LEFT` with `IF_UNCONS`, etc

**Review by michaeljklein — APPROVED** [review]
LGTM: minor comment on loop vs. fold but not necessary

File: packages/minter-contracts/ligo/src/swaps/fa2_allowlisted_swap_offchain_nocounter.mligo (hunk 1 above)
Comment (EGuenz) [reply packages/minter-contracts/ligo/src/swaps/fa2_allowlisted_swap_offchain_nocounter.mligo]: Fixed!

**Review by michaeljklein — APPROVED** [review]
LGTM

**Pull request merged**